### PR TITLE
UAR-640: Added MOs JSON

### PR DESCRIPTION
**Pull request by bbell-companieshouse**
### JIRA link
https://companieshouse.atlassian.net/browse/UAR-640

### Change description
Implement MO additions into the update submission model

### Work checklist

- [ ] Tests added where applicable

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/changelist/additions/CorporateManagingOfficerAddition.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
+package uk.gov.companieshouse.overseasentitiesapi.model.updatesubmission.changelist.additions;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import uk.gov.companieshouse.overseasentitiesapi.model.updatesubmission.changelist.commonmodels.Address;
+import uk.gov.companieshouse.overseasentitiesapi.model.updatesubmission.changelist.commonmodels.CompanyIdentification;
+
+import java.time.LocalDate;
+
+public class CorporateManagingOfficerAddition extends ManagingOfficerAddition{
```
Comment (AndreKramer7654) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/changelist/additions/CorporateManagingOfficerAddition.java:9]: space before {

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/ManagingOfficerAdditionService.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+                        addressDtoToAddress(serviceAddress),
+                        resignedOn);
+
+        individualManagingOfficerAddition.setPersonName(new PersonName(mo.getFirstName(), mo.getLastName()));
+        individualManagingOfficerAddition.setFormerNames(mo.getFormerNames());
+        individualManagingOfficerAddition.setBirthDate(mo.getDateOfBirth());
+        individualManagingOfficerAddition.setRole(mo.getRoleAndResponsibilities());
+        individualManagingOfficerAddition.setOccupation(mo.getOccupation());
+
+        if (Objects.isNull(mo.getSecondNationality())) {
```
Comment (AndreKramer7654) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/ManagingOfficerAdditionService.java:60]: We may be doing this in several places so worth pulling out a utility method? Just the test and optional concat.

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/ManagingOfficerAdditionService.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        corporateManagingOfficerAddition.setName(mo.getName());
+        corporateManagingOfficerAddition.setContactName(mo.getContactFullName());
+        corporateManagingOfficerAddition.setContactEmail(mo.getContactEmail());
+
+        var legalForm = mo.getLegalForm();
+        var governingLaw = mo.getLawGoverned();
+        var registerLocation = mo.getPublicRegisterName();
+        var registrationNumber = mo.getRegistrationNumber();
+        var identification = new CompanyIdentification(
+                legalForm, governingLaw, registerLocation, null, registrationNumber);
```
Comment (AndreKramer7654) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/ManagingOfficerAdditionService.java:103]: seems strange that the null is second last. Maybe registerLocation and registrationNumber need to be beside each other in construct arg list? Would need refactoring of other usages.

**Review by AndreKramer7654 — APPROVED** [review]

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/changelist/additions/IndividualManagingOfficerAddition.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
+package uk.gov.companieshouse.overseasentitiesapi.model.updatesubmission.changelist.additions;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import uk.gov.companieshouse.overseasentitiesapi.model.updatesubmission.changelist.commonmodels.Address;
+import uk.gov.companieshouse.overseasentitiesapi.model.updatesubmission.changelist.commonmodels.PersonName;
+
+import java.time.LocalDate;
+
+public class IndividualManagingOfficerAddition extends ManagingOfficerAddition{
```
Comment (n-r-b) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/changelist/additions/IndividualManagingOfficerAddition.java:9]: According to the mapping documentation, we need to set change=officerAppointmentChange for both individual and corporate MOs

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/changelist/additions/IndividualManagingOfficerAddition.java [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    @JsonProperty("personName")
+    private PersonName personName;
+
+    @JsonProperty("formerNames")
+    private String formerNames;
```
Comment (n-r-b) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/changelist/additions/IndividualManagingOfficerAddition.java:14]: Should this be an array of strings?

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/changelist/additions/CorporateManagingOfficerAddition.java [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    @JsonProperty("name")
+    private String name;
+
+    @JsonProperty("contactName")
+    private String contactName;
+
+    @JsonProperty("contactEmail")
+    private String contactEmail;
+
+    @JsonProperty("identification")
```
Comment (n-r-b) [inline src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/changelist/additions/CorporateManagingOfficerAddition.java:19]: According to the documentation
```suggestion
    @JsonProperty("companyIdentification")
```

File: src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/changelist/additions/IndividualManagingOfficerAddition.java (hunk 5 above)
Comment (bbell-companieshouse) [reply src/main/java/uk/gov/companieshouse/overseasentitiesapi/model/updatesubmission/changelist/additions/IndividualManagingOfficerAddition.java:14]: I believe its just a string taken from ManagingOfficerIndividualDto.formerNames

**Review by n-r-b — APPROVED** [review]

**Pull request merged**